src/material/tabs/tab-group.html

```diff
@@ -51,6 +51,16 @@
   }
 </mat-tab-header>
 
+<!--
+  We need to project the content somewhere to avoid hydration errors. Some observations:
+  1. This is only necessary on the server.
+  2. We get a hydration error if there aren't any nodes after the `ng-content`.
+  3. We get a hydration error if `ng-content` is wrapped in another element.
+-->
+@if (_isServer) {
+  <ng-content/>
```

Comment:
why is this needed? I don't follow and would like to understand more. Why isn't this needed with CSR?

Reply:
Ohh, I see what you're saying. I would have expected the `mat-tab` to be resolved as well and have it's content SSR-ed as well. That seems like a surprising requirement for projection?

Reply:
It is a bit weird, but it's kinda expected. If the `mat-tab` doesn't get projected, when the HTML makes it to the browser, it won't know that it existed when it was created on the server.

Reply:
It feels surprising to me that component authors would need to take care of that. It almost sounds like a bug to me that it works in CSR like that (talking about `ContentChildren` matching without a ng-content`).

The change makes sense, but would love the comment to capture this as well, but no need to re-push. Also cc. @AndrewKushnir as some user-facing feedback on hydration potentially